docs/root/configuration/http/http_filters/_include/credential-injector-oauth2-filter.yaml

```diff
@@ -0,0 +1,79 @@
+static_resources:
+  listeners:
+  - name: listener_0
+    address:
+      socket_address:
+        address: 0.0.0.0
+        port_value: 10000
+    filter_chains:
+    - filters:
+      - name: envoy.filters.network.http_connection_manager
+        typed_config:
+          "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
+          stat_prefix: ingress_http
+          route_config:
+            name: local_route
+            virtual_hosts:
+            - name: local_service
+              domains: ["*"]
+              routes:
+              - match:
+                  prefix: "/"
+                route:
+                  cluster: local-backend-service
+          http_filters:
+          - name: envoy.filters.http.credential_injector
+            typed_config:
+              "@type": type.googleapis.com/envoy.extensions.filters.http.credential_injector.v3.CredentialInjector
+              credential:
+                name: envoy.http.injected_credentials.oauth2
+                typed_config:
+                  "@type": type.googleapis.com/envoy.extensions.http.injected_credentials.oauth2.v3.OAuth2
+                  token_endpoint:
+                    cluster: okta.ad
+                    timeout: 3s
+                    uri: "https://dev-1178504991.okta.com/oauth2/default/v1/token"
+                  client_credentials:
+                    client_id: some-client-id
+                    client_secret:
+                      name: client-secret
+          - name: envoy.filters.http.router
+            typed_config:
+              "@type": type.googleapis.com/envoy.extensions.filters.http.router.v3.Router
```

Comment:
nit: can we add a blank line before `clusters` - it makes it easier to jump around the file